src/VecSim/algorithms/hnsw/hnsw_wrapper.cpp

```diff
@@ -112,8 +114,19 @@ int HNSWIndex::addVector(const void *vector_data, size_t id) {
 
 int HNSWIndex::deleteVector(size_t id) {
     bool res = this->hnsw->removePoint(id);
-    if (hnsw->getIndexSize() + this->blockSize <= this->hnsw->getIndexCapacity()) {
-        this->hnsw->resizeIndex(this->hnsw->getIndexCapacity() - this->blockSize);
+    size_t index_size = hnsw->getIndexSize();
+    size_t block_size = this->blockSize;
+    size_t curr_capacity = this->hnsw->getIndexCapacity();
+
+    // if we need to free a complete block & there is a least one block between the
+    // capacity and the size
+    if (index_size % block_size == 0 && index_size + block_size <= curr_capacity) {
```

Comment:
this can happen if the capacity is at least twice the block size,
if the condition will be 
index_size % block_size != 0 && index_size + block_size <= curr_capacity
each deleteion will cause the resizing of the index, until the gap is reduced

Reply:
I understand now that this is intentional. Sounds good